src/Splat/ServiceLocation/ModernDependencyResolver.cs

```diff
@@ -142,7 +139,7 @@ public IEnumerable<object> GetServices(Type serviceType, string? contract = null
                 return Enumerable.Empty<object>();
             }
 
-            return _registry[pair].Select(x => x()!).ToList();
+            return _registry[pair].ConvertAll(x => x()!);
```

Comment:
There's some roslynator or netanalyzers checks that can find these Linq 2 Optimised versions btw out there. Might be worth activating them for all the projects but not necessary for this PR (and I will do it rather than dumping it all on you).

Reply:
We already have the analysers installed, that's how I found them.